---
bool lock_released
uint32 counter
